cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/transform/ScriptTransform.java

```diff
@@ -167,31 +184,46 @@ private StructuredRecord decodeRecord(Map nativeObject, Schema schema) {
     return builder.build();
   }
 
+  private List jsObject2List(Object object) {
+    if (object.getClass().getName().equals("jdk.nashorn.api.scripting.ScriptObjectMirror")) {
```

Comment:
doing a null check would be better than string comparison if we know that's the correct behavior.

it's weird that they represent a json array as a map...